android/autodispose-android-archcomponents/src/main/java/com/uber/autodispose/android/lifecycle/KotlinExtensions.kt

```diff
@@ -91,92 +91,74 @@ inline fun Lifecycle.scope(
 ): ScopeProvider = AndroidLifecycleScopeProvider.from(
     this, boundaryResolver)
 
-/**
- * Extension that proxies to [Flowable.as] + [AutoDispose.autoDisposable]
- */
-@CheckReturnValue
-inline fun <T> Flowable<T>.autoDisposable(lifecycleOwner: LifecycleOwner): FlowableSubscribeProxy<T> =
-    this.`as`(AutoDispose.autoDisposable(AndroidLifecycleScopeProvider.from(lifecycleOwner)))
-
 /**
  * Extension that proxies to [Flowable.as] + [AutoDispose.autoDisposable] and takes an [untilEvent] when
  * subscription will be disposed.
  */
 @CheckReturnValue
-inline fun <T> Flowable<T>.autoDisposable(lifecycleOwner: LifecycleOwner, untilEvent: Event): FlowableSubscribeProxy<T> =
-    this.`as`(AutoDispose.autoDisposable(AndroidLifecycleScopeProvider.from(lifecycleOwner, untilEvent)))
-
-/**
- * Extension that proxies to [Observable.as] + [AutoDispose.autoDisposable]
- */
-@CheckReturnValue
-inline fun <T> Observable<T>.autoDisposable(lifecycleOwner: LifecycleOwner): ObservableSubscribeProxy<T> =
-    this.`as`(AutoDispose.autoDisposable(AndroidLifecycleScopeProvider.from(lifecycleOwner)))
+inline fun <T> Flowable<T>.autoDisposable(lifecycleOwner: LifecycleOwner, untilEvent: Event? = null): FlowableSubscribeProxy<T> {
+  untilEvent?.let {
```

Comment:
nit: let's just keep this a simple `if (untilEvent == null)` and extract the `return` to the top with the if statement as the expression

Reply:
That's what I had before but then I was like maybe I should be more Kotlin-y 😄

Reply:
38de073